File: AzDG.java
import java.io.UnsupportedEncodingException;
import java.util.*;
import java.security.*; 
	
public class AzDg { 
	private static String strPrivateKey = "Private key";
	
	public static byte[] encode(byte[] argAbteSource, byte[] argAbteEncryptKey) {
		argAbteEncryptKey = new Md5().getMD5ofStr(new String(argAbteEncryptKey)).toLowerCase().getBytes();
		byte bteCRCLength = 0;  
		byte[] abteReturn = new byte[argAbteSource.length];
		for (int i = 0; i < argAbteSource.length; ++i) {
			bteCRCLength = (bteCRCLength > 31) ? 0 : bteCRCLength;
			abteReturn[i] = (byte)(argAbteSource[i] ^ argAbteEncryptKey[bteCRCLength++]);
		}
		return abteReturn;
	}
	
	public static String encrypt(String argStrSource) {
  		return encrypt(argStrSource, strPrivateKey);
 	}
	
 	public static String encrypt(String argStrSource, String argStrPrivateKey) {
  		return encrypt(argStrSource, argStrPrivateKey,"UTF8");
 	}

 	public static String encrypt(String argStrSource, String argStrPrivateKey, String argStrCharset) {
  		String strReturn = null;
  		try	{
   			strReturn = new String(encrypt(argStrSource.getBytes(argStrCharset), argStrPrivateKey.getBytes(argStrCharset)),argStrCharset);
  		}
  		catch (UnsupportedEncodingException e) {
   			e.printStackTrace();
  		}
 		return strReturn;
 	}

 	public static byte[] encrypt(byte[] argStrSource, byte[] argBteKey) {  		
  		byte[] abteEncryptKey = new Md5().getMD5ofStr(Long.toString( Calendar.getInstance().getTimeInMillis())).getBytes();
  		byte bteCRCLength = 0;  
  		byte[] abteReturn = new byte[argStrSource.length * 2];  
		for (int i = 0, j = 0; i < argStrSource.length; ++i, ++j) {
			bteCRCLength = bteCRCLength > 31 ? 0 : bteCRCLength;  
			abteReturn[j] = abteEncryptKey[bteCRCLength];
			++j;
			abteReturn[j] = (byte)(argStrSource[i] ^ abteEncryptKey[bteCRCLength++]);
		}		
  		return Base64.encode(encode(abteReturn, argBteKey));
	}
 	
 	public static String decrypt(String argStrSource) {
 		return decrypt(argStrSource,strPrivateKey);
	}
 	
	public static String decrypt(String argStrSource, String argStrPrivateKey)	{
		return decrypt(argStrSource, argStrPrivateKey, "UTF8");
	}
	
	public static String decrypt(String argStrSource, String argStrPrivateKey, String argStrCharset) {
		String strReturn ="";
		try {
			strReturn= new String(decrypt(Base64.decode(argStrSource), argStrPrivateKey.getBytes(argStrCharset)),argStrCharset);
		}
		catch (UnsupportedEncodingException e) {
   			e.printStackTrace();
  		}
 		return strReturn;
	}

	public static byte[] decrypt(byte[] argStrSource, byte[] argStrPrivateKey) {
		argStrSource = encode(argStrSource, argStrPrivateKey);
		byte[] abteReturn = new byte[(int)(argStrSource.length / 2)]; 
		for (int i = 0, j = 0; i < argStrSource.length; ++i, ++j) {
			abteReturn[j] = (byte)(argStrSource[i] ^ argStrSource[++i] );
		}  
		return abteReturn;
	}
		
	public static byte[] md5Byte(String d) throws NoSuchAlgorithmException {
		MessageDigest md = MessageDigest.getInstance("MD5"); 		
		md.update(d.getBytes()); 
		return  md.digest(); 		
	}		
}

File: AzDG.cs
//.Net3.5
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace jIAnSoft.Framework.Security.Cryptography {
    public class AzDG {
        private static string _strCharSet = "utf-8";
        private const string StrCompanyKey = "Private key";
        private static string _strGenuineKey = "";

        private static byte[] PrivateKeyCrypt(IList<byte> argAbteSource) {
            var abteEncryptKey = Encoding.GetEncoding(_strCharSet).GetBytes(Md5.Encrypt(_strGenuineKey).ToLower());
            var abteReturn = new byte[argAbteSource.Count];
            for (var i = 0; i < argAbteSource.Count; ++i) {
                abteReturn[i] = (byte)(argAbteSource[i] ^ abteEncryptKey[i % 32]);
            }
            return abteReturn;
        }

        private static byte[] DoEncrypt(IList<byte> argStrSource) {
            var abteCrcKey =
                Encoding.GetEncoding(_strCharSet).GetBytes(
                    Md5.Encrypt(
                        DateTime.Now.Ticks.ToString(
                            CultureInfo.InvariantCulture)).ToLower());
            var abteReturn = new byte[argStrSource.Count * 2];
            for (var j = 0; j < argStrSource.Count; j++) {
                abteReturn[(j * 2)] = abteCrcKey[j % 32];
                abteReturn[(j * 2) + 1] = (byte)(argStrSource[j] ^ abteCrcKey[j % 32]);
            }
            return PrivateKeyCrypt(abteReturn);
        }

        public static String Encrypt(String argStrSource, String argStrPrivateKey = "") {
            return Encrypt(argStrSource, argStrPrivateKey, _strCharSet);
        }

        public static String Encrypt(String argStrSource, String argStrPrivateKey, String argStrCharset) {
            _strCharSet = argStrCharset;
            _strGenuineKey = string.Format("{0}{1}", StrCompanyKey, argStrPrivateKey);
            return Convert.ToBase64String(DoEncrypt(Encoding.GetEncoding(_strCharSet).GetBytes(argStrSource)));
        }

        private static byte[] DoDecrypt(byte[] argStrSource) {
            argStrSource = PrivateKeyCrypt(argStrSource);
            var abteReturn = new byte[argStrSource.Length / 2];
            for (var j = 0; j < abteReturn.Length; ++j) {
                abteReturn[j] = (byte)(argStrSource[(j * 2)] ^ argStrSource[(j * 2) + 1]);
            }
            return abteReturn;
        }

        public static String Decrypt(String argStrSource, String argStrPrivateKey = "") {
            return Decrypt(argStrSource, argStrPrivateKey, _strCharSet);
        }

        public static String Decrypt(String argStrSource, String argStrPrivateKey, String argStrCharset) {
            _strGenuineKey = string.Format("{0}{1}", StrCompanyKey, argStrPrivateKey);
            return Encoding.GetEncoding(argStrCharset).GetString(DoDecrypt(Convert.FromBase64String(argStrSource)));
        }
    }
}
